end2end-test-examples/gcs/src/main/java/io/grpc/gcs/GcsioClient.java

```diff
@@ -202,11 +174,36 @@ private void makeRandomMediaRequest(ResultTable results) throws IOException {
       if (buff.remaining() > 0) {
         logger.warning("Got remaining bytes: " + buff.remaining());
       }
-      logger.info("time cost for reading bytes: " + dur + "ms");
-      results.reportResult(dur);
+      results.reportResult(args.bkt, object, args.buffSize * 1024, dur);
     }
     reader.close();
 
     gcsfs.close();
   }
+
+  private void makeWriteRequest(ResultTable results, int threadId)
+      throws IOException, InterruptedException {
+    GoogleCloudStorageFileSystem gcsfs = new GoogleCloudStorageFileSystem(creds,
+        GoogleCloudStorageFileSystemOptions.builder().setCloudStorageOptions(gcsOpts).build());
+
+    int size = args.size * 1024;
+    Random rd = new Random();
+    byte[] randBytes = new byte[size];
+    rd.nextBytes(randBytes);
+
+    for (int i = 0; i < args.calls; i++) {
+      long start = System.currentTimeMillis();
+      String object = objectResolver.Resolve(threadId, i);
+      URI uri = URI.create("gs://" + args.bkt + "/" + object);
+      WritableByteChannel writeChannel = gcsfs.create(uri);
+      ByteBuffer buff = ByteBuffer.wrap(randBytes);
+      writeChannel.write(buff);
+      writeChannel.close();
+      // write operation is async, need to call close() to wait for finish.
```

Comment:
Should this comment be moved before L207?

Reply:
> Could you move this comment before 201?
Done.

> And should L197 to L202 be put inside the for loop or outside?
What do you mean by that?